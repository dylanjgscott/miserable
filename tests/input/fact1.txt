FUNCTION factorial(n)
VARS cond, tmp;
BEGIN
	cond = (n == 0);
	IF cond THEN
	BEGIN
		tmp = 1;
	END
	ELSE
	BEGIN
		tmp = (n - 1);
		tmp = (factorial(tmp) * n);
	END;
    RETURN tmp;
END

FUNCTION main(n)
VARS tmp;
BEGIN
	tmp = factorial(n);
	RETURN tmp;
END
