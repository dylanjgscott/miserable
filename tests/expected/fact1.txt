[Function "factorial" (Args ["n"]) (Vars ["cond","tmp"]) [Assign "cond" (ExpOp OpEq (ExpId "n") (ExpNum 0)),IfElse "cond" [Assign "tmp" (ExpNum 1),Return "tmp"] [Assign "tmp" (ExpOp OpSub (ExpId "n") (ExpNum 1)),Assign "tmp" (ExpOp OpMul (ExpFun "factorial" (Args ["tmp"])) (ExpId "n")),Return "tmp"]],Function "main" (Args ["n"]) (Vars ["tmp"]) [Assign "tmp" (ExpFun "factorial" (Args ["n"])),Return "tmp"]]
